packages/core/src/sync/events.ts

```diff
@@ -78,8 +148,15 @@ export const decodeEvents = (
             topics: event.log!.topics,
           });
 
+          if (
+            event.log!.topics[0] === undefined ||
+            source.abiEvents.bySelector[event.log!.topics[0]] === undefined
+          ) {
+            throw new Error();
+          }
```

Comment:
What does this guard against?

Reply:
This guards against us getting a random value for topic0 back from the rpc. This doesn't actually change the behavior of this function, it just more explicitly throws an error here rather than accessing a property on undefined